src/js/background/context-menu.js

```diff
@@ -153,15 +172,42 @@ const relayContextMenus = {
         reuseAliasMenuIdPrefix,
         ""
       );
-
       // Get stored alias data
-      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      const relayAddresses = await relayContextMenus.utils.getAliases();
 
       // Select the correct alias from the stored alias data
       const selectedAliasObject = relayAddresses.filter((alias) => {
         return alias.id === parseInt(selectedAliasId, 10);
       });
 
+      const serverStoragePref = await getCachedServerStoragePref();
+
+    
+      const currentMaskType = selectedAliasObject[0].mask_type;
+      const currentUsedOnValue = selectedAliasObject[0].used_on;
+    
+      const currentPage = new URL(tab.url);
+      const currentPageHostName = currentPage.hostname;
+ 
+      // If the used_on field is blank, then just set it to the current page/hostname. Otherwise, add/check if domain exists in the field
+      const used_on = (currentUsedOnValue === null || currentUsedOnValue === "" ||  currentUsedOnValue === undefined)
+        ? `${currentPageHostName},` : relayContextMenus.utils.addUsedOnDomain(currentUsedOnValue, currentPageHostName);
```

Comment:
Maybe the logic described in the comment could be moved into `addUsedOnDomain`? I think it'll be relevant every time it gets called, no?

Reply:
Ahhh. Great callout. Migrated logic (and renamed the function to `checkAndStoreUsedOnDomain()`) in both the context menu and the in-page menu scripts.

Revised in cf7b8922020cc1f389f341b503d3362b4cfa4132